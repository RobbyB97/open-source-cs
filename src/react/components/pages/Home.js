/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import Description from '../sections/Description';


/* Component */
export class Home extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('Home');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="Home" className="Home">
                <header className="Home__header">
                    <h1 className="Home__title">
                        The Open Source Computer Science degree
                    </h1>
                </header>
                <Description />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(Home);