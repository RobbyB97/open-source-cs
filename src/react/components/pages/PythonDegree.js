/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import CSBasicsCourses from '../../data/python-sections/CSBasicsCourses';
import ProgrammingCourses from '../../data/python-sections/ProgrammingCourses';
import ProgrammingAlternateCourses from '../../data/python-sections/ProgrammingAlternateCourses';
import MathCourses from '../../data/python-sections/MathCourses';
import DataScienceCourses from '../../data/python-sections/DataScienceCourses';
import SystemsCourses from '../../data/python-sections/SystemsCourses';
import TheoryCourses from '../../data/python-sections/TheoryCourses';
import ApplicationsCourses from '../../data/python-sections/ApplicationsCourses';
import UnixCourses from '../../data/python-sections/UnixCourses';


/* Component */
export class PythonDegree extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('PythonDegree');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="PythonDegree" className="PythonDegree">
                <header className="PythonDegree__header">
                    <h1>
                        Python Degree program
                    </h1>
                </header>

                <CourseSection 
                    name="CS Basics"
                    courses={CSBasicsCourses}
                />

                <CourseSection 
                    name="Programming"
                    courses={ProgrammingCourses}
                />

                <CourseSection 
                    name="Programming (alternative)"
                    courses={ProgrammingAlternateCourses}
                />

                <CourseSection 
                    name="Math"
                    courses={MathCourses}
                />

                <CourseSection 
                    name="Data Science"
                    courses={DataScienceCourses}
                />

                <CourseSection 
                    name="Systems"
                    courses={SystemsCourses}
                />

                <CourseSection 
                    name="Theory"
                    courses={TheoryCourses}
                />

                <CourseSection 
                    name="Applications"
                    courses={ApplicationsCourses}
                />

                <CourseSection 
                    name="Unix"
                    courses={UnixCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(PythonDegree);
