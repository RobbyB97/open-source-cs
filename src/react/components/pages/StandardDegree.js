/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Actions */
import {page_ID__Set} from '../../redux/actions/page';


/* Components */
import CourseSection from '../sections/CourseSection';


/* Data */
import ApplicationsCourses from '../../data/standard-sections/ApplicationsCourses';
import CSBasicsCourses from '../../data/standard-sections/CSBasicsCourses';
import MathCourses from '../../data/standard-sections/MathCourses';
import ProgrammingCourses from '../../data/standard-sections/ProgrammingCourses';
import SystemsCourses from '../../data/standard-sections/SystemsCourses';
import TheoryCourses from '../../data/standard-sections/TheoryCourses';
import UnixCourses from '../../data/standard-sections/UnixCourses';


/* Component */
export class StandardDegree extends React.Component {
    constructor(props) {
        super(props);
    }


    componentWillMount() {
        this.page_ID__Set('StandardDegree');
    }


    page_ID__Set = (id) => {
        this.props.page_ID__Set(id);
    }


    render() {
        return (
            <div id="StandardDegree" className="StandardDegree">
                <header className="StandardDegree__header">
                    <h1>
                        Standard Degree program
                    </h1>
                </header>

                <CourseSection 
                    name="CS Basics"
                    courses={CSBasicsCourses}
                />

                <CourseSection 
                    name="Programming"
                    courses={ProgrammingCourses}
                />

                <CourseSection 
                    name="Math"
                    courses={MathCourses}
                />

                <CourseSection 
                    name="Systems"
                    courses={SystemsCourses}
                />

                <CourseSection 
                    name="Theory"
                    courses={TheoryCourses}
                />

                <CourseSection 
                    name="Applications"
                    courses={ApplicationsCourses}
                />

                <CourseSection 
                    name="Unix"
                    courses={UnixCourses}
                />
            </div>
        );
    }
}


/* Connect to store */
const mapDispatchToProps = (dispatch) => ({
    page_ID__Set: (id) => {
        dispatch(page_ID__Set(id));
    }
});


export default connect(undefined, mapDispatchToProps)(StandardDegree);
