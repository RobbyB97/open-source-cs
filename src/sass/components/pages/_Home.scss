.Home {
    @include page;
}


.Home__header {
    @include pageHeader;
    text-align: center;
}
