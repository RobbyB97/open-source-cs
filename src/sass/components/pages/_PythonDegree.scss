.PythonDegree {
    @include page;
}


.PythonDegree__header {
    @include pageHeader;
    text-align: center;
}