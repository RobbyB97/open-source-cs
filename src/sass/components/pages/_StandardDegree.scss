.StandardDegree {
    @include page;
}


.StandardDegree__header {
    @include pageHeader;
    text-align: center;
}