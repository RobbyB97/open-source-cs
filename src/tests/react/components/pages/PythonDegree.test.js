/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {PythonDegree} from '../../../../react/components/pages/PythonDegree';


/* Test Config */
let component, page_ID__Set;
beforeEach(() => {
    page_ID__Set = jest.fn();
    component = shallow(
        <PythonDegree
            page_ID__Set={page_ID__Set}
        />
    );
});


/* Tests */
test('Render PythonDegree page', () => {
    expect(component).toMatchSnapshot();
});


test('Page_ID__Set function', () => {
    setTimeout(() => {
        expect(page_ID__Set).toHaveBeenCalled();
    }, 500);
});

