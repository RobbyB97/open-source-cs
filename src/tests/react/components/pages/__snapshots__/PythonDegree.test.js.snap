// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PythonDegree page 1`] = `
<div
  className="PythonDegree"
  id="PythonDegree"
>
  <header
    className="PythonDegree__header"
  >
    <h1>
      Python Degree program
    </h1>
  </header>
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "10",
          "effort": "10-20",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/cs50s-introduction-to-computer-science",
            "text": "Intro to Computer Science",
          },
          "school": "Harvard",
        },
        Object {
          "duration": "6",
          "effort": "2-5",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/what-is-a-proof",
            "text": "Mathematical Thinking in Computer Science",
          },
          "school": "UC San Diego",
        },
      ]
    }
    name="CS Basics"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "4",
          "effort": "4-6",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python",
            "text": "Programming for everybody (Getting Started with Python)",
          },
          "school": "University of Michigan",
        },
        Object {
          "duration": "4",
          "effort": "4-6",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-data",
            "text": "Python Data Structures",
          },
          "prerequisites": Array [
            "Programming for Everybody (Getting Started with Python)",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "4",
          "effort": "4-6",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-network-data",
            "text": "Using Python to Access Web Data",
          },
          "prerequisites": Array [
            "Python Data Structures",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "3",
          "effort": "4-8",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-databases",
            "text": "Using Databases with Python",
          },
          "prerequisites": Array [
            "Using Python to Access Web Data",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "2",
          "effort": "4-6",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-data-visualization",
            "text": "Capstone: Retrieving, Processing and Visualizing Data with Python",
          },
          "prerequisites": Array [
            "Using Databases with Python",
          ],
          "school": "University of Michigan",
        },
      ]
    }
    name="Programming"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "20",
          "effort": "7",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/specializations/python-3-programming",
            "text": "Python 3 Programming Specialization",
          },
          "school": "University of Michigan",
        },
      ]
    }
    name="Programming (alternative)"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "15",
          "effort": "9-10",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/precalculus",
            "text": "Precalculus",
          },
          "school": "ASU",
        },
        Object {
          "duration": "12",
          "effort": "6-10",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/calculus-1a-differentiation",
            "text": "Calculus 1A: Differentiation",
          },
          "prerequisites": Array [
            "Precalculus",
          ],
          "school": "MIT",
        },
        Object {
          "duration": "15",
          "effort": "6-10",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/calculus-1b-integration",
            "text": "Calculus 1B: Integration",
          },
          "prerequisites": Array [
            "Calculus 1A: Differentiation",
          ],
          "school": "MIT",
        },
        Object {
          "duration": "8",
          "effort": "6-10",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/calculus-1c-coordinate-systems-infinite-series",
            "text": "Calculus 1C: Coordinate Systems & Infinite Series",
          },
          "prerequisites": Array [
            "Calculus 1B: Integration",
          ],
          "school": "MIT",
        },
        Object {
          "duration": "15",
          "effort": "6-10",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.edx.org/course/linear-algebra-foundations-to-frontiers",
            "text": "Linear Algebra - Foundations to Frontiers",
          },
          "prerequisites": Array [
            "Precalculus",
          ],
          "school": "UT Austin",
        },
        Object {
          "duration": "5",
          "effort": "5-7",
          "frequency": "Twice a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/probability-intro",
            "text": "Introduction to Probability and Data",
          },
          "school": "Duke",
        },
        Object {
          "duration": "8",
          "effort": "5-7",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.udacity.com/course/intro-to-statistics--st101",
            "text": "Intro to Statistics",
          },
          "school": "Udacity",
        },
      ]
    }
    name="Math"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "3",
          "effort": "4-6",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-data-analysis",
            "text": "Introduction to Data Science in Python",
          },
          "prerequisites": Array [
            "Basic programming knowledge",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "4",
          "effort": "5-7",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-plotting",
            "text": "Applied Plotting, Charting & Data Representation in Python",
          },
          "prerequisites": Array [
            "Introduction to Data Science in Python",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "6",
          "effort": "5-7",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-machine-learning",
            "text": "Applied Machine Learning in Python",
          },
          "prerequisites": Array [
            "Applied Plotting, Charting & Data Representation in Python",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "5",
          "effort": "5-7",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-text-mining",
            "text": "Applied Text Mining in Python",
          },
          "prerequisites": Array [
            "Applied Machine Learning in Python",
          ],
          "school": "University of Michigan",
        },
        Object {
          "duration": "5",
          "effort": "5-7",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.coursera.org/learn/python-social-network-analysis",
            "text": "Applied Social Network Analysis in Python",
          },
          "prerequisites": Array [
            "Applied Text Mining in Python",
          ],
          "school": "University of Michigan",
        },
      ]
    }
    name="Data Science"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "6",
          "effort": "5",
          "frequency": "Twice a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/build-a-computer",
            "text": "Build a Modern Computer from First Principles: From Nand to Tetris",
          },
          "prerequisites": Array [
            "Basic programming knowledge",
          ],
          "school": "Hebrew University of Jerusalem",
        },
        Object {
          "duration": "6",
          "effort": "10-15",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/nand2tetris2",
            "text": "Build a Modern Computer from First Principles: From Nand to Tetris II",
          },
          "prerequisites": Array [
            "Build a Modern Computer from First Principled: From Nand to Tetris",
          ],
          "school": "Hebrew University of Jerusalem",
        },
        Object {
          "duration": "8",
          "effort": "5-8",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.udacity.com/course/introduction-to-operating-systems--ud923",
            "text": "Introduction to Operating Systems",
          },
          "prerequisites": Array [
            "Build a Modern Computer from First Principles: From Nand to Tetris II",
          ],
          "school": "Georgia Tech",
        },
      ]
    }
    name="Systems"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "10",
          "effort": "2-5",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/cs-algorithms-theory-machines",
            "text": "Computer Science: Algorithms, Theory and Machines",
          },
          "prerequisites": Array [
            "Calculus 1A",
            "Calculus 1B",
            "Calculus 1C",
            "Basic programming knowledge",
          ],
          "school": "Princeton",
        },
        Object {
          "duration": "6",
          "effort": "6-12",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/algorithms-part1",
            "text": "Algorithms, Part I",
          },
          "prerequisites": Array [
            "Computer Science: Algorithms, Theory and Machines",
          ],
          "school": "Princeton",
        },
        Object {
          "duration": "6",
          "effort": "6-12",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/algorithms-part2",
            "text": "Algorithms, Part II",
          },
          "prerequisites": Array [
            "Algorithms, Part II",
          ],
          "school": "Princeton",
        },
      ]
    }
    name="Theory"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "7",
          "effort": "4-6",
          "frequency": "Twice a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/database-management",
            "text": "Database Management Essentials",
          },
          "prerequisites": Array [
            "Basic programming & CS knowledge",
          ],
          "school": "CU",
        },
        Object {
          "duration": "7",
          "effort": "5",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/crypto",
            "text": "Cryptography I",
          },
          "prerequisites": Array [
            "Linear Algebra - Foundations to Frontiers",
            "Introduction to Probability and Data",
          ],
          "school": "Stanford",
        },
      ]
    }
    name="Applications"
  />
  <CourseSection
    courses={
      Array [
        Object {
          "duration": "1",
          "effort": "5",
          "frequency": "Self-paced",
          "name": Object {
            "link": "https://www.udacity.com/course/linux-command-line-basics--ud595",
            "text": "Linux Command Line Basics",
          },
          "school": "Udacity",
        },
        Object {
          "duration": "4",
          "effort": "4",
          "frequency": "Once a month",
          "name": Object {
            "link": "https://www.coursera.org/learn/unix",
            "text": "The Unix Workbench",
          },
          "school": "JHU",
        },
      ]
    }
    name="Unix"
  />
</div>
`;
